### PR TITLE
DOC: adding operational form documentation for array ops

### LINKED ISSUE #17384: basic array operators are hard to find in the docs
## Documentation

Documentation on [basic operations](https://numpy.org/doc/stable/user/quickstart.html#basic-operations) is hard to find. It would be very useful if associated operators were mentioned in the documentation of method equivalents. 

For example, element-wise logical operations such as `np.logical_or(a, b)` have an operator equivalent `a | b`. The operational form, however, is not mentioned in the [`np.logical_or`](https://numpy.org/doc/stable/reference/generated/numpy.logical_or.html) documentation nor in the [summary page](https://numpy.org/doc/stable/reference/routines.logic.html) of logic functions. In fact, I've searched for it in numpy documentation a number of times over the years, and always have difficulty finding it.

### PR DESCRIPTION
**Pull request by Rashmi-K-A**
xref gh-17384

---

* [x] ``np.less`` &mdash; ``<``
* [x] ``np.less_equal`` &mdash; ``<=``
* [x] ``np.equal`` &mdash; ``==``
* [x] ``np.not_equal`` &mdash; ``!=``
* [x] ``np.greater`` &mdash; ``>``
* [x] ``np.greater_equal`` &mdash; ``>=``
* [x] ``np.add`` &mdash; ``+``
* [x] ``np.subtract`` &mdash; ``-``
* [x] ``np.multiply`` &mdash; ``*``
* [x] ``np.matmul`` &mdash; ``@``
* [x] ``np.true_divide`` &mdash; ``/``
* [x] ``np.floor_divide`` &mdash; ``//``
* [x] ``np.remainder`` &mdash; ``%``
* [x] ``np.divmod`` &mdash; ``divmod(a, b)``
* [x] ``np.power`` &mdash; ``**``
* [x] ``np.left_shift`` &mdash; ``<<``
* [x] ``np.right_shift`` &mdash; ``>>``
* [x] ``np.bitwise_and`` &mdash; ``&``
* [x] ``np.bitwise_xor`` &mdash; ``^``
* [x] ``np.bitwise_or`` &mdash; ``|``
* [x] ``np.negative`` &mdash; ``-``
* [x] ``np.positive`` &mdash; ``+``
* [x] ``np.absolute`` &mdash; ``abs(a)``
* [x] ``np.invert`` &mdash; ``~``

### CODE REVIEW

File: numpy/core/code_generators/ufunc_docstrings.py
Comment (eric-wieser) [inline numpy/core/code_generators/ufunc_docstrings.py:2267]: This one is incorrect - `&` is `np.bitwise_and`, which happens to behave the same on booleans as `np.logical_and`. You can prove that with
```python
class A(np.ndarray):
    def __array_ufunc__(self, ufunc, method, *args, **kwargs):
        print(ufunc)

np.array(1).view(A) & True
```
which gives
```
<ufunc 'bitwise_and'>
```

It's still worth mentioning `&` here, but noting that it's only appropriate to use on boolean arrays.

File: numpy/core/code_generators/ufunc_docstrings.py
Comment (eric-wieser) [inline numpy/core/code_generators/ufunc_docstrings.py:2343]: As above, this is `bitwise_or`

**Comment by eric-wieser** [comment]
Thanks for this, this looks like a good way to start addressing that issue.

File: numpy/core/code_generators/ufunc_docstrings.py
Comment (eric-wieser) [inline numpy/core/code_generators/ufunc_docstrings.py:1126]: I wonder if we want to invert this phrasing to something like

```suggestion
    The `==` operator on arrays is implemented using ``np.equal``
```

or


```suggestion
    The `==` operator on arrays is a shorthand for ``np.equal``
```

What I want to get across is that actually `np.equal` is the real deal, and `==` is a shorthand. Will defer to @bjnath and @rossbar who probably can come up with a better wording

File: numpy/core/code_generators/ufunc_docstrings.py
Comment (Rashmi-K-A) [reply numpy/core/code_generators/ufunc_docstrings.py:2343]: I'll make the changes to these both functions.

**Comment by eric-wieser** [comment]
I think you're missing a few operators here. You should be able to check against these lines, which contain a full list of the numpy functions (written with `um.` instead of `np.`) against the python operator overloads:

https://github.com/numpy/numpy/blob/57adb4bb6a8eb46fed597d4a781e47bcb86ebe11/numpy/lib/mixins.py#L140-L176

File: numpy/core/code_generators/ufunc_docstrings.py
Comment (Rashmi-K-A) [reply numpy/core/code_generators/ufunc_docstrings.py]: I haven't got any further comments about better wording so I'll wait on this for now. The PR is still in draft for now.

**Comment by eric-wieser** [comment]
I've added a checklist at the top of this post for all the operators that we should add - would you mind checking off the ones you've done so far?

**Comment by Rashmi-K-A** [comment]
I have checked the ones done, will add the missing ones. np.greater's operational form documentation was already there so am checking that as well.

File: numpy/core/code_generators/ufunc_docstrings.py
Comment (rossbar) [reply numpy/core/code_generators/ufunc_docstrings.py]: I like @eric-wieser 's "implemented" suggestion, but either suggested wording is good IMO

Comment (Rashmi-K-A) [reply numpy/core/code_generators/ufunc_docstrings.py]: "Is a shorthand for" does sound better than "is implemented using". Does something along the lines of "The `==` operator can be used as a shorthand for``np.equal`` on ndarrays" sound ok?

Comment (eric-wieser) [reply numpy/core/code_generators/ufunc_docstrings.py]: That sounds fine to me. @seberg, can you think of a wording that still leaves us open to adding a note that there are some cases when `==` is better than `equal` (due to what I remember as broken behavior on string/void arrays)?

Comment (seberg) [reply numpy/core/code_generators/ufunc_docstrings.py]: String/void is simply not implemented for `np.equal` (it would be possible, but annoying right now). There are some other (usually with deprecation warning), differences.  I thought about it for a while, and honestly do not have a good idea of how to try and achieve more precision. Unless we say "(see Notes for details)"?

Comment (Rashmi-K-A) [reply numpy/core/code_generators/ufunc_docstrings.py]: Have we reached a conclusion on this?

Comment (rossbar) [reply numpy/core/code_generators/ufunc_docstrings.py]: It sounds to me like your latest wording suggestion ("The == operator can be used as a shorthand fornp.equal on ndarrays") is sufficient. Since there are no obvious technical problems with the wording, my vote would be to push ahead with the suggested wording. It can always be revisited for tweaks/improvements later on.

**Comment by Rashmi-K-A** [comment]
@eric-wieser  is there anything else required? Wondering because this PR isn't approver or merged.

**Comment by Rashmi-K-A** [comment]
@BvB93 I noticed you linked this PR to the main issue a couple of weeks back, I haven't got any further notice about this PR, neither any comments nor changes if any. Could you let me know how to proceed with this one?

**Review by BvB93 — COMMENTED** [review]
Overall I've got two style-related comments. Could you take a look at them?

File: numpy/core/code_generators/ufunc_docstrings.py
Comment (Rashmi-K-A) [reply numpy/core/code_generators/ufunc_docstrings.py]: I made these changes 👍

**Review by BvB93 — APPROVED** [review]

**Review by rossbar — APPROVED** [review]
I went ahead and made a couple minor changes (mostly related to style) to some of the examples, but generally this LGTM. @Rashmi-K-A please take a look at the changes and make sure everything still looks okay to you. I'll aim to put this in today, thanks @Rashmi-K-A !

**Comment by Rashmi-K-A** [comment]
Looks good to me, thanks! @rossbar

**Pull request merged**
